what-is-chalk.mdx

```diff
@@ -18,215 +18,467 @@ import { PyDiffEditor } from '@/components/Editor'
+
+Chalk connects directly to your existing infrastructure and underlying data stores (Athena, Bigquery, Postgres, Iceberg catalog, etc.) with [SQL resolvers](https://docs.chalk.ai/docs/sql-resolvers) and to APIs (microservices, 3rd part clients, LLMs) with [Python resolvers.](https://docs.chalk.ai/docs/python-resolvers)
+Resolvers make it easy to integrate a wide variety of data sources, join them together, and use them in inference.
+
+Decoupling feature logic from ETL pipelines unlocks
+
+- faster iteration cycles
+- on-demand just-in-time inference
+- reproducible and testable features (with point-in-time accuracy)
 
 <IntegrationTypes />
 
----
+### How does Chalk orchestrate and manage features?
 
-## Execute
+Chalk has built-in support for feature engineering workflows — there's no need to manage Airflow or orchestrate complicated streaming flows.
+Once you've defined features and resolvers, Chalk orchestrates them into flexible pipelines (slicing the DAG) that make both training and model execution easy.
 
-Once you've defined your features and resolvers, Chalk orchestrates
-them into flexible pipelines that make training and executing models easy.
+The `get_username` function below explicitly defines input dependencies through `User.email` and specifies the output feature type as `User.username`:
 
-Chalk has built-in support for feature engineering workflows --
-there's no need to manage Airflow or orchestrate complicated streaming flows.
-You can execute resolver pipelines with declarative caching,
-ingest batch data on a schedule, and easily make slow sources
-available online for low-latency serving.
+```py
+@online
+def get_username(
+    # the input type annotation specifies the required feature
+    email: User.email,
+    # the resulting feature is saved as username unto User class
+) -> User.username:
+    username = email.split("@")[0]
+    if "gmail.com" in email:
+        username = username.split("+")[0].replace(".", "")
+
+    return username.lower()
+```
+
+Although written in Python, this function compiles into a static Velox expression through Chalk's Symbolic Python Interpreter and executes natively in C++.
+This provides Python-first development without sacrificing sub-5ms inference latency.
+
+```py
+lower(
+    if_then_else(
+        !=(
+            strpos(str(email), str(gmail.com)),
+            int(0)
+        ),
+        replace(
+            element_at(
+                split(
+                    element_at(split(str(email), str(@)), int(1)),
+                    str(+)
+                ),
+                int(1)
+            ),
+            str(.),
+            str()
+        ),
+        element_at(split(str(email), str(@)), int(1))
+    )
+)
+```
+
+We wrote an [engineering blog post on our Symbolic Python interpreter](https://chalk.ai/blog/symbolic-python-interpreter) if you want to learn more!
 
-### Caching
+### How can I cache features in Chalk?
```

Comment:
The `I` is a little odd to me in the context of the rest of the doc -- it shows up only here and in the next header